TopQuarkAnalysis/TopEventProducers/src/PseudoTopProducer.cc

```diff
@@ -0,0 +1,578 @@
+#include "TopQuarkAnalysis/TopEventProducers/interface/PseudoTopProducer.h"
+
+#include "CommonTools/Utils/interface/PtComparator.h"
+
+#include "RecoJets/JetProducers/interface/JetSpecific.h"
+#include "fastjet/ClusterSequence.hh"
+
+using namespace std;
+using namespace edm;
+using namespace reco;
+
+PseudoTopProducer::PseudoTopProducer(const edm::ParameterSet& pset):
+  leptonMinPt_(pset.getParameter<double>("leptonMinPt")),
+  leptonMaxEta_(pset.getParameter<double>("leptonMaxEta")),
+  jetMinPt_(pset.getParameter<double>("jetMinPt")),
+  jetMaxEta_(pset.getParameter<double>("jetMaxEta")),
+  wMass_(pset.getParameter<double>("wMass")),
+  tMass_(pset.getParameter<double>("tMass"))
+{
+  finalStateToken_ = consumes<edm::View<reco::Candidate> >(pset.getParameter<edm::InputTag>("finalStates"));
```

Comment:
If I had to critique something, then it would be that not all class member variables are initialised in the initialisation list (finalStateToken_ ,genParticleToken_ ,fjLepDef_ , fjJetDef_ , genVertex_ ).
Is this something special regarding the ```consumes``` keyword (not familiar with that one)?
If so, I would at least recommend the implementation of the destructor for these variabels.